libqtile/widget/base.py

```diff
@@ -311,12 +312,17 @@ def calculate_length(self):
 
     def timeout_add(self, seconds, method, method_args=()):
         """
-            This method calls either ``.call_later`` with given arguments.
+            This method calls ``.call_later`` with given arguments.
         """
-        self.future = self.qtile.call_later(
+        future = self.qtile.call_later(
             seconds, self._wrapper, method, *method_args
         )
 
+        # Remove completed timers from the list and add this timer
+        self._futures = [timer for timer in self._futures if timer._scheduled]
```

Comment:
Rather than clearing a timer when the next is set up, could we clear it when it finishes i.e. as part of the `timer_setup` logic? Or would this require all users (widgets, including custom widgets) to remember and do it manually?

Reply:
This is more about stopping a memory leak. For example, the `Clock` widget creates a timer every second. Without this clean up, the `_futures` list would grow and grow.

Reply:
Why could this not be done at the end of a task, rather than the start of the next?

Reply:
We could include it in the wrapper for the callback. Happy to make that change.

Reply:
I think that would be better - it would ensure that futures are removed from the list even if there is no task following it.

Reply:
The only downside to this is that if every timer is cancelled, the `_futures` list will grow and grow. Pretty unlikely scenario though!

Reply:
What can cause a timer to get cancelled, other than when Qtile exits?

Reply:
Badly written code! 

I'm just pushing the change now.